app/hooks/useAppLoading.ts

```diff
@@ -15,28 +15,49 @@ import {
 } from '../store/slices/wallet';
 import { getAllRecords } from '../store';
 import { useAppDispatch } from './useAppDispatch';
+import { loadRemoteRegistries } from '../lib/registry';
 
 export function useAppLoading(): boolean {
-  const dispatch = useAppDispatch();
   const [loading, setLoading] = useState(true);
 
-  const { isUnlocked, isInitialized } = useSelector(selectWalletState);
+  const primaryTasks = [
+    useFontsLoaded(),
+    useWalletStateInitialized(),
+  ];
+
+  const primaryTasksFinished = useMemo(() => primaryTasks.every(t => t), primaryTasks);
+
+  useEffect(() => { 
+    if (primaryTasksFinished) runSecondaryTasks();
+  }, [primaryTasksFinished]);
+
+  async function runSecondaryTasks() {
+    await Promise.all([
+      loadRemoteRegistries(),
+    ]);
+
+    setLoading(false);
+  }
+
+  return loading;
+}
 
+function useFontsLoaded() {
   const [fontsLoaded] = useFonts({
     Rubik_400Regular,
     Rubik_500Medium,
     Rubik_700Bold,
     Roboto_400Regular,
   });
 
-  const walletStateInitialized = isUnlocked !== null && isInitialized !== null;
-  const finishedLoading = walletStateInitialized && fontsLoaded;
+  return fontsLoaded;
+}
 
-  useEffect(() => {
-    if (finishedLoading) {
-      setLoading(false);
-    }
-  }, [finishedLoading]);
+function useWalletStateInitialized() {
+  const dispatch = useAppDispatch();
+
+  const { isUnlocked, isInitialized } = useSelector(selectWalletState);
+  const walletStateInitialized = isUnlocked !== null && isInitialized !== null;
```

Comment:
```suggestion
  const walletStateInitialized = isUnlocked && isInitialized;
```

Reply:
@dmitrizagidulin I'm not sure updating the logic here to use boolean type assertion allows the state initialization logic to remain intact. The values for variables across different app states goes as follows:
| State | `isInitialized` | `isUnlocked ` |
| --- | --- | --- |
| Initial values | `null` | `null` |
| Start screen | `false` | `false` |
| Home screen | `true` | `true` |
| Login screen | `true` | `false` |